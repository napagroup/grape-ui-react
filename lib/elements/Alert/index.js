"use strict";

var _Object$keys = require("@babel/runtime-corejs3/core-js-stable/object/keys");

var _forEachInstanceProperty = require("@babel/runtime-corejs3/core-js-stable/instance/for-each");

var _context, _context2;

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

var _component = require("./component");

_forEachInstanceProperty(_context = _Object$keys(_component)).call(_context, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _component[key];
    }
  });
});

var _styled = require("./styled");

_forEachInstanceProperty(_context2 = _Object$keys(_styled)).call(_context2, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _styled[key];
    }
  });
});