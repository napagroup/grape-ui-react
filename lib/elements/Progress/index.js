"use strict";

var _Object$keys = require("@babel/runtime-corejs3/core-js-stable/object/keys");

var _forEachInstanceProperty = require("@babel/runtime-corejs3/core-js-stable/instance/for-each");

var _context, _context2;

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

var _getProgress = require("./getProgress");

_forEachInstanceProperty(_context = _Object$keys(_getProgress)).call(_context, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _getProgress[key];
    }
  });
});

var _styled = require("./styled");

_forEachInstanceProperty(_context2 = _Object$keys(_styled)).call(_context2, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _styled[key];
    }
  });
});