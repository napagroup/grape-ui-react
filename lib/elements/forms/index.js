"use strict";

var _Object$keys = require("@babel/runtime-corejs3/core-js-stable/object/keys");

var _forEachInstanceProperty = require("@babel/runtime-corejs3/core-js-stable/instance/for-each");

var _context, _context2, _context3, _context4, _context5, _context6;

var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");

_Object$defineProperty(exports, "__esModule", {
  value: true
});

var _CheckboxField = require("./CheckboxField");

_forEachInstanceProperty(_context = _Object$keys(_CheckboxField)).call(_context, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _CheckboxField[key];
    }
  });
});

var _DateField = require("./DateField");

_forEachInstanceProperty(_context2 = _Object$keys(_DateField)).call(_context2, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _DateField[key];
    }
  });
});

var _Form = require("./Form");

_forEachInstanceProperty(_context3 = _Object$keys(_Form)).call(_context3, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _Form[key];
    }
  });
});

var _RadioField = require("./RadioField");

_forEachInstanceProperty(_context4 = _Object$keys(_RadioField)).call(_context4, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _RadioField[key];
    }
  });
});

var _SelectField = require("./SelectField");

_forEachInstanceProperty(_context5 = _Object$keys(_SelectField)).call(_context5, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _SelectField[key];
    }
  });
});

var _TextField = require("./TextField");

_forEachInstanceProperty(_context6 = _Object$keys(_TextField)).call(_context6, function (key) {
  if (key === "default" || key === "__esModule") return;

  _Object$defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _TextField[key];
    }
  });
});