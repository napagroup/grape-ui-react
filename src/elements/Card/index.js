export * from './component';
export * from './styled';
