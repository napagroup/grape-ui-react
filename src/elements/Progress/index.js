export * from './getProgress';
export * from './styled';
