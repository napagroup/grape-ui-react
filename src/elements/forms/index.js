export * from './CheckboxField';
export * from './DateField';
export * from './Form';
export * from './SelectField';
export * from './TextField';
