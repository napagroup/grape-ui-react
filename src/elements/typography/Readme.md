### "Typography is the craft of endowing human language with a durable visual form.”

&mdash; Robert Bringhurst, [The Elements of Typographic Style](https://www.amazon.com/Elements-Typographic-Style-Version-Anniversary/dp/088179211X/ref=dp_ob_title_bk)

Typography is important in grape-ui.  Here is an example using the typography components in grape-ui based on [an example from Simon Griffee](https://www.simongriffee.com/typography/)

```jsx in Markdown
import {
  Code,
  CodeBlock,
  Header,
  Link,
  List,
  ListItem,
  Paragraph,
  Text,
} from '/';

const urlSimonGriffee = "https://www.simongriffee.com/design/earth-among-the-stars/";

const urlStarMaker = "http://www.astro.sunysb.edu/fwalter/AST389/TEXTS/StarMaker.pdf";

<div>
  <Header>
    The Earth
    <br />
    <Text italic>
      The Starting Point
    </Text>
  </Header>
  <Paragraph lead>
    One night when I had tasted bitterness I went out on to the hill. Dark heather checked my feet. Below marched the suburban lamps.
  </Paragraph>
  <Paragraph>
    {'Windows, their curtains drawn, were shut eyes, inwardly watching the lives of dreams. Beyond the sea’s level darkness a lighthouse pulsed. Overhead, obscurity. I distinguished our own house, our islet in the tumultuous and bitter currents of the world. There, for a decade and a half, we two, so different in quality, had grown in and in to one another, for mutual support and nourishment, in intricate symbiosis. There daily we planned our several undertakings, and recounted the day’s oddities and vexations. There letters piled up to be answered, socks to be darned. There the children were born, those sudden new lives. There, under that roof, our own two lives, recalcitrant sometimes to one another, were all the while thankfully one, one larger, more '}
    <Link
      href={urlStarMaker}
    >
      conscious life
    </Link>
    {' than either alone.'}
  </Paragraph>
  <Header.h2>
    All This, Surely, Was Good
  </Header.h2>
  <Paragraph>
    Yet there was bitterness. And bitterness not only invaded us from the world; it welled up also within our own magic circle. For horror at our futility, at our own unreality, and not only at the world’s delirium, had driven me out on to the hill.
  </Paragraph>
  <Paragraph>
    We were always hurrying from one little urgent task to another, but the upshot was insubstantial. Had we, perhaps, misconceived our whole existence? Were we, as it were, living from false premises? And in particular, this partnership of ours, this seemingly so well-based fulcrum for activity in the world, was it after all nothing but a little eddy of complacent and ingrown domesticity, ineffectively whirling on the surface of the great flux, having in itself no depth of being, and no significance? Had we perhaps after all deceived ourselves? Behind those rapt windows did we, like so many others, indeed live only a dream? In a sick world even the hale are sick. And we two, spinning our little life mostly by rote, seldom with clear cognizance, seldom with firm intent, were products of a sick world.
  </Paragraph>
  <Paragraph>
    Yet this life of ours was not all sheer and barren fantasy. Was it not spun from the actual fibres of reality, which we gathered in with all the comings and goings through our door, all our traffic with the suburb and the city and with remoter cities, and with the ends of the earth? And were we not spinning together an authentic expression of our own nature? Did not our life issue daily as more or less firm threads of active living, and mesh itself into the growing web, the intricate, ever-proliferating pattern of mankind?
  </Paragraph>
  <Header.h3>
    I Considered “Us” With Quiet Interest
  </Header.h3>
  <Paragraph>
    And a kind of amused awe. How could I describe our relationship even to myself without either disparaging it or insulting it with the tawdry decoration of sentimentality? For this our delicate balance of dependence and independence, this coolly critical, shrewdly ridiculing, but loving mutual contact, was surely a microcosm of true community, was after all in its simple style an actual and living example of that high goal which the world seeks.
  </Paragraph>
  <blockquote>
    <Paragraph>
      {'I reflected that not one of the visible features of this celestial and living gem revealed the presence of man. Displayed before me, though invisible, were some of the most congested centers of human population. There below me lay huge industrial regions, blackening the air with smoke. Yet all this thronging life and humanly momentous enterprise had made no mark whatever on the features of the planet. From this high look-out the Earth would have appeared no different before the dawn of man. No visiting angel, or explorer from another planet, could have guessed that this bland orb teemed with vermin, with world-mastering, self-torturing, incipiently angelic beasts. '}
      <Text sm>
        {'— Olaf Stapledon, '}
        <cite title="Star Maker">
          Star Maker
        </cite>
      </Text>
    </Paragraph>
  </blockquote>
  <Paragraph>
    The whole world? The whole universe? Overhead, obscurity unveiled a star. One tremulous arrow of light, projected how many thousands of years ago, now stung my nerves with vision, and my heart with fear. For in such a universe as this what significance could there be in our fortuitous, our frail, our evanescent community?
  </Paragraph>
  <Header.h4>
    But Now Irrationally I Was Seized
  </Header.h4>
  <Paragraph>
    With a strange worship, not, surely of the star, that mere furnace which mere distance falsely sanctified, but of something other, which the dire contrast of the star and us signified to the heart. Yet what, what could thus be signified? Intellect, peering beyond the star, discovered no Star Maker, but only darkness; no Love, no Power even, but only Nothing.
  </Paragraph>
  <Header.h5>
    And Yet the Heart Praised
  </Header.h5>
  <Paragraph>
    Impatiently I shook off this folly, and reverted from the inscrutable to the familiar and the concrete. Thrusting aside worship, and fear also and bitterness, I determined to examine more coldly this remarkable “us,” this surprisingly impressive datum, which to ourselves remained basic to the universe, though in relation to the stars it appeared so slight a thing.
  </Paragraph>
  <Header.h6>
    Considered Even Without Reference to Our Belittling Cosmical Background
  </Header.h6>
  <Paragraph>
    We were after all insignificant, perhaps ridiculous. We were such a commonplace occurrence, so trite, so respectable. We were just a married couple, making shift to live together without undue strain. Marriage in our time was suspect. And ours, with its trivial romantic origin, was doubly suspect.
  </Paragraph>
  <List>
    <ListItem>
      We had first met when she was a child
    </ListItem>
    <ListItem>
      Our eyes encountered
    </ListItem>
    <ListItem>
      She looked at me for a moment with quiet attention
    </ListItem>
    <ListItem>
      Even, I had romantically imagined, with obscure, deep-lying recognition
      <List>
        <ListItem>
          I, at any rate, recognized in that look
        </ListItem>
        <ListItem>
          So I persuaded myself in my fever of adolescence
        </ListItem>
      </List>
    </ListItem>
    <ListItem>
      My destiny
    </ListItem>
  </List>
  <Paragraph>
    <Text fontWeight="bold">
      {'Yes! '}
    </Text>
    <Text italic>
      How predestinate had seemed our union!
    </Text>
    {' Yet now, in retrospect, how accidental. True, '}
    <del>
      of course
    </del>
    {', that as a long-married couple we '}
    <kbd>
      fitted
    </kbd>
    {' rather neatly, like '}
    <Text sm>
      two close trees whose trunks
    </Text>
    {' have grown upwards together as a single shaft, mutually distorting, but mutually supporting. '}
    <abbr title="abbreviation title">
      ABC
    </abbr>
  </Paragraph>
  <List.ol>
    <ListItem>
      The Diversity of Worlds
    </ListItem>
    <ListItem>
      Strange Mankinds
    </ListItem>
    <ListItem>
      Nautiloids
    </ListItem>
    <ListItem>
      The Earth
      <List>
        <ListItem>
          The Starting Point
        </ListItem>
        <ListItem>
          Earth Among the Stars
        </ListItem>
      </List>
    </ListItem>
    <ListItem>
      More Worlds
    </ListItem>
  </List.ol>

  <Paragraph>
    {`Here's some code: `}
    <Code code="e=mc²" />
  </Paragraph>
  <CodeBlock
    code={`
      And
      Some
      Poetry
    `}
  />
  <dl>
    <dt>
      Coldly
    </dt>
    <dd>
      I now assessed her as merely a useful, but often infuriating adjunct to my personal life.
    </dd>
    <dt>
      We
    </dt>
    <dd>
      Were on the whole sensible companions.
    </dd>
    <dd>
      We left one another a certain freedom, and so we were able to endure our proximity.
    </dd>
    <dt>
      Such
    </dt>
    <dd>
      Was our relationship.
    </dd>
  </dl>
  <hr />
  <Paragraph sm>
    {'Text excerpt from '}
    <Link
      href={urlStarMaker}
    >
      <Text italic>
        Star Maker
      </Text>
      {' by Olaf Stapledon'}
    </Link>
    <br />
    <Link
      href="https://gitlab.com/simongriffee/typography"
    >
      Free template
    </Link>
    {' by '}
    <Link
      href={urlSimonGriffee}
    >
      Simon Griffee
    </Link>
    {'.'}
  </Paragraph>
</div>
```
