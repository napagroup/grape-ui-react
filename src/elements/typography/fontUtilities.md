Here are some more utilities that could be used:

* `ellipsis`
* `fontWeight`
* `italic`
* `kerning`
* `textAlign`
* `textDecoration`

```jsx in Markdown
import {
  Code,
  Paragraph,
  Text,
} from './index';

<div>
  <Paragraph
    ellipsis
  >
    {'Set '}
    <Code code="ellipsis" />
    {' to true if you want the sentence to end in a single line. Set '}
    <Code code="ellipsis" />
    {' to true if you want the sentence to end in a single line.'}
  </Paragraph>
  <Paragraph
    ellipsis={2}
  >
    {'Set '}
    <Code code="ellipsis" />
    {' to a number if you want the sentence to end at N line.'}
    {'Set '}
    <Code code="ellipsis" />
    {' to a number if you want the sentence to end at N line.'}
    {'Set '}
    <Code code="ellipsis" />
    {' to a number if you want the sentence to end at N line.'}
    {'Set '}
    <Code code="ellipsis" />
    {' to a number if you want the sentence to end at N line.'}
  </Paragraph>
  <Paragraph>
    {'Add '}
    <Code code="fontWeight" />
    {' to make text '}
    <Text
      fontWeight="300"
    >
      Different
    </Text>
    {' '}
    <Text
      fontWeight="500"
    >
      font
    </Text>
    {' '}
    <Text
      fontWeight="700"
    >
      weights
    </Text>
    {'.  Just define the value.'}
  </Paragraph>
  <Paragraph>
    {'Add '}
    <Code code="italic" />
    {' to make text '}
    <Text italic>
      Italic
    </Text>
  </Paragraph>
  <Paragraph>
    {'Add '}
    <Code code="kerning" />
    {' to adjust the spacing '}
    <Text
      kerning="1px"
    >
      between
    </Text>
    {' '}
    <Text
      kerning="0.25rem"
    >
      letters
    </Text>
    .  Just use any numeric value to adjust.
  </Paragraph>
  <Paragraph>
    {'Add '}
    <Code code="textAlign" />
    {' to align any text element. Just use any '}
    <Code code="text-align" language="css" />
    {' property as the value.'}
  </Paragraph>
  <Paragraph
    textAlign="left"
  >
    Left Aligned
    <br />
    <Code code={'textAlign="left"'} />
    {' is set.'}
  </Paragraph>
  <Paragraph
    textAlign="center"
  >
    Center Aligned
    <br />
    <Code code={'textAlign="center"'} />
    {' is set.'}
  </Paragraph>
  <Paragraph
    textAlign="right"
  >
    Right Aligned
    <br />
    <Code code={'textAlign="right"'} />
    {' is set.'}
  </Paragraph>
  <Paragraph>
    {'Add '}
    <Code code="textDecoration" />
    {' to set '}
    <Text
      textDecoration="underline"
    >
      one
    </Text>
    {' or '}
    <Text
      textDecoration="line-through"
    >
      {' more '}
    </Text>
    <Text
      textDecoration="underline overline"
    >
      text-decorations.
    </Text>
  </Paragraph>
</div>
```
