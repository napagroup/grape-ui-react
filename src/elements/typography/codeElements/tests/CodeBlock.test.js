import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { Code } from 'src/elements/typography';

const emptyTheme = {};

const assertReactElement = element => {
  const component = renderer.create(element);
  return component.toJSON();
};

describe('Code Component base', () => {
  it('should return an empty Code with base styling', () => {
    const element = (
      <ThemeProvider theme={emptyTheme}>
        <Code code={'<Code code="" />'} />
      </ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });
  it('should return an empty Code with a valid JSX element', () => {
    const element = (
      <ThemeProvider theme={emptyTheme}>
        <Code code={<Code code={'<Code code="" />'} />} />
      </ThemeProvider>
    );
    expect(assertReactElement(element)).toMatchSnapshot();
  });
});
