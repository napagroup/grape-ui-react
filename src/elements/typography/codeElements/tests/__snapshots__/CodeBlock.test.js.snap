// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Component base should return an empty Code with base styling 1`] = `
.c0 {
  background: whitesmoke;
  border: 1px solid #eee;
  border-radius: 4px;
  background-color: transparent;
  color: inherit;
  font-size: 80%;
  font-weight: inherit;
  display: inline-block;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  margin-top: 0;
  margin-bottom: 0;
  padding: 4px;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c0 > code {
  font-family: Consolas,"Andale Mono WT","Andale Mono","Lucida Console","Lucida Sans Typewriter","DejaVu Sans Mono","Bitstream Vera Sans Mono","Liberation Mono","Nimbus Mono L",Monaco,"Courier New",Courier,monospace;
}

<pre
  className="c0"
  style={
    Object {
      "background": "hsl(0, 0%, 98%)",
    }
  }
>
  <code>
    &lt;Code code=
    <span
      style={
        Object {
          "color": "hsl(174.4, 100%, 29.4%)",
        }
      }
    >
      ""
    </span>
     /&gt;
  </code>
</pre>
`;
